core/modules/nat.h

```diff
@@ -155,6 +167,10 @@ class NAT final : public Module {
 
   std::vector<be32_t> ext_addrs_;
 
+  // Port ranges available for each address. The first index is the same as the
+  // ext_addrs_ range.
+  std::vector<std::list<PortRange>> port_ranges_;
```

Comment:
How about vector in place of list, as per Effective STL (vectors should be used by default for read-mostly sequences)?